Actually it looks like we have to move materialx to python3Packages to fix this?
```
$ nix-build -A python311.pkgs.openusd
...
-- Found Boost: /nix/store/w4dfa1akyciynv6i9l0963fr386wbayk-boost-1.81.0-dev/lib/cmake/Boost-1.81.0/BoostConfig.cmake (found version "1.81.0")
-- Disabling boost-provided cmake config
-- Found Python3: /nix/store/z7xxy35k7620hs6fn6la5fg2lgklv72l-python3-3.12.4/bin/python3.12 (found version "3.12.4") found components: Interpreter Development Development.Module Development.Embed
CMake Error at /nix/store/ll2b6cslly4v30im7qk34f9y21ziiz1i-cmake-3.29.6/share/cmake-3.29/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost (missing: python312) (found version "1.81.0")
```

--- a/pkgs/applications/misc/blender/default.nix
+++ b/pkgs/applications/misc/blender/default.nix
@@ -16,7 +16,6 @@
   config,
   cudaPackages,
   cudaSupport ? config.cudaSupport,
-  darwin,
   dbus,
   embree,
   fetchpatch,
@@ -55,7 +54,6 @@
   libxkbcommon,
   llvmPackages,
   makeWrapper,
-  materialx,
   mesa,
   ocl-icd,
   openal,
@@ -127,7 +125,7 @@ stdenv.mkDerivation (finalAttrs: {
           substituteInPlace source/creator/CMakeLists.txt \
             --replace-fail '${"$"}{LIBDIR}/python' \
                       '${python3}' \
-            --replace-fail '${"$"}{LIBDIR}/materialx/' '${materialx}/'
+            --replace-fail '${"$"}{LIBDIR}/materialx/' '${python3Packages.materialx}/'
           substituteInPlace build_files/cmake/platform/platform_apple.cmake \
             --replace-fail '${"$"}{LIBDIR}/brotli/lib/libbrotlicommon-static.a' \
                       '${lib.getLib brotli}/lib/libbrotlicommon.dylib' \
@@ -160,7 +158,7 @@ stdenv.mkDerivation (finalAttrs: {
       "-DWITH_FFTW3=ON"
       "-DWITH_IMAGE_OPENJPEG=ON"
       "-DWITH_INSTALL_PORTABLE=OFF"
-      "-DMaterialX_DIR=${materialx}/lib/cmake/MaterialX"
+      "-DMaterialX_DIR=${python3Packages.materialx}/lib/cmake/MaterialX"
       "-DWITH_MOD_OCEANSIM=ON"
       "-DWITH_OPENCOLLADA=${if colladaSupport then "ON" else "OFF"}"
       "-DWITH_OPENCOLORIO=ON"
@@ -233,7 +231,6 @@ stdenv.mkDerivation (finalAttrs: {
       libsndfile
       libtiff
       libwebp
-      materialx
       opencolorio
       openexr
       openimageio
@@ -244,6 +241,7 @@ stdenv.mkDerivation (finalAttrs: {
       potrace
       pugixml
       python3
+      python3Packages.materialx
       tbb
       zlib
       zstd
@@ -299,7 +297,7 @@ stdenv.mkDerivation (finalAttrs: {
       ps = python3Packages;
     in
     [
-      materialx
+      ps.materialx
       ps.numpy
       ps.requests
       ps.zstandard

--- a/pkgs/development/python-modules/materialx/default.nix
+++ b/pkgs/development/python-modules/materialx/default.nix
@@ -1,19 +1,20 @@
 {
   lib,
   stdenv,
+  buildPythonPackage,
   fetchFromGitHub,
   cmake,
+  setuptools,
   darwin,
   libX11,
   libXt,
   libGL,
   openimageio,
   imath,
-  python3Packages,
-  python3
+  python,
 }:
 
-python3Packages.buildPythonPackage rec {
+buildPythonPackage rec {
   pname = "materialx";
   version = "1.38.10";
 
@@ -28,7 +29,7 @@ python3Packages.buildPythonPackage rec {
 
   nativeBuildInputs = [
     cmake
-    python3Packages.setuptools
+    setuptools
   ];
 
   buildInputs =
@@ -60,7 +61,7 @@ python3Packages.buildPythonPackage rec {
 
   postInstall = ''
     # Make python lib properly accessible
-    target_dir=$out/${python3.sitePackages}
+    target_dir=$out/${python.sitePackages}
     mkdir -p $(dirname $target_dir)
     # required for cmake to find the bindings, when included in other projects
     ln -s $out/python $target_dir

--- a/pkgs/development/python-modules/materialx/default.nix
+++ b/pkgs/development/python-modules/materialx/default.nix
@@ -22,7 +22,7 @@ buildPythonPackage rec {
     owner = "AcademySoftwareFoundation";
     repo = "MaterialX";
     rev = "v${version}";
-    sha256 = "sha256-/kMHmW2dptZNtjuhE5s+jvPRIdtY+FRiVtMU+tiBgQo=";
+    hash = "sha256-/kMHmW2dptZNtjuhE5s+jvPRIdtY+FRiVtMU+tiBgQo=";
   };
 
   format = "other";
@@ -68,6 +68,7 @@ buildPythonPackage rec {
   '';
 
   meta = {
+    changelog = "https://github.com/AcademySoftwareFoundation/MaterialX/blob/${src.rev}/CHANGELOG.md";
     description = "Open standard for representing rich material and look-development content in computer graphics";
     homepage = "https://materialx.org";
     maintainers = [ lib.maintainers.gador ];

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5735,6 +5735,8 @@ with pkgs;
 
   mat2 = with python3.pkgs; toPythonApplication mat2;
 
+  materialx = with python3Packages; toPythonApplication materialx;
+
   maxcso = callPackage ../tools/archivers/maxcso { };
 
   measureme = callPackage ../development/tools/rust/measureme { };

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -7344,6 +7344,8 @@ self: super: with self; {
 
   material-color-utilities = callPackage ../development/python-modules/material-color-utilities { };
 
+  materialx = callPackage ../development/python-modules/materialx { };
+
   matchpy = callPackage ../development/python-modules/matchpy { };
 
   mathlibtools = callPackage ../development/python-modules/mathlibtools { };

